app/scripts/modules/core/src/pipeline/config/triggers/artifacts/bitbucket/BitbucketArtifactEditor.tsx

```diff
@@ -42,27 +42,39 @@ export const BitbucketDefault: IArtifactKindConfig = {
     }
 
     private onReferenceChanged = (reference: string) => {
-      const pathRegex = new RegExp('/1.0/repositories/[^/]*/[^/]*/raw/[^/]*/(.*)$');
-      const results = pathRegex.exec(reference);
-      if (results !== null) {
-        const clonedArtifact = cloneDeep(this.props.artifact);
-        clonedArtifact.name = decodeURIComponent(results[1]);
-        clonedArtifact.reference = reference;
-        this.props.onChange(clonedArtifact);
-      }
+      const clonedArtifact = cloneDeep(this.props.artifact);
+      clonedArtifact.reference = reference;
+      this.props.onChange(clonedArtifact);
+    };
+
+    private onFilePathChanged = (name: string) => {
+      const clonedArtifact = cloneDeep(this.props.artifact);
+      clonedArtifact.name = name;
+      this.props.onChange(clonedArtifact);
```

Comment:
Small nit on re-usability here - You can turn this into one method and parameterize the method to accept the field you want to update and the value that it should be updated with. Here's an example:

```
private onArtifactPropChanged = (field: string, value: string) => {
  const clonedArtifact = cloneDeep(this.props.artifact);
  clonedArtifact[field] = value;
  this.props.onChange(clonedArtifact);
}
```

and then when It's being used on the component, for example, the `name` field,  you can do something like this:
```
onChange={(name: string) => this.onArtifactPropChanged('name', name)}
```